version: "3.5"
services:
  postgres:
    image: postgres:14-alpine
    container_name: postgres
    environment:
      POSTGRES_USER: "admin"
      POSTGRES_PASSWORD: "test"
      POSTGRES_DB: "taskforce-tasks"
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U admin -d test" ]
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s
    volumes:
      - ./db:/var/lib/postgresql/data
    networks:
      - postgres

  pgadmin:
    container_name: pgadmin
    image: dpage/pgadmin4
    environment:
      PGADMIN_DEFAULT_EMAIL: "test@taskforce.local"
      PGADMIN_DEFAULT_PASSWORD: "test"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - ./db:/var/lib/pgadmin
    ports:
      - "8082:80"
    restart: unless-stopped
    networks:
      - postgres

  taskforce.tasks:
    image: taskforce.tasks
    depends_on:
      - postgres
      - pgadmin
    container_name: taskforce.tasks.service
    ports:
      - "3335:3333"
    env_file:
      - ../../environments/.tasks.env

networks:
  postgres:
    driver: bridge
